### DISCUSSION TITLE
Bottom Up/Top Down vs Bottom Up-ID/Top Down-ID

### DISCUSSION THREAD

**Discussion by djg9324** [talmolab/sleap#1569] [Help!] [answered]
I have been doing multi-animal tracking and realized that the program is not doing so well with distinguishing which mouse is which. I have experimented with Top Down training and am now starting to experiment with Bottom Up training. However, I did want to try to use the new id versions of the two. When I do, I receive an error due to memory storage on my GUI. What I would like to understand is what are the differences between the regular training versions and the id training versions. Also, how exactly does the id versions conflict with the computer's memory storage so much. What suggestions are there to help my sleap model distinguish my two mice better?

**Comment by roomrys** [comment, marked as answer]
Hi @djg9324,

The ID models train another network to classify animals into `Track`s based on the additional `Track` annotations you've added - thus, they require more compute. Whenever you run into Out Of Memory (OOM) issues, the suggestion is usually the same for each model (IF you want to stick to using that model - top-down is the most memory efficient):
1. **Lower the input scaling** (for top-down you can only adjust the input scaling for the centroid model - the centered instance model always keeps/expects the cropped image at full resolution). How low can you go? Zoom in on your smallest labeled body part. How many pixels wide/tall is it. We can lower the input scaling s.t. when you multiply input scaling by both the height and width of that body part there will still be 2 pixels in both the scaled height and width. This ensures that we don't accidentally erase that body part when scaling too low. Input scaling will have the most affects on memory.
2. **Decrease the batch size**. Batch size is the other parameter that can cause OOM issues. The less data that is input into the network at a time, the less needs to be kept in memory.

Thanks,
Liezl
